### PR TITLE
[READY]Fixes flightsuits without making the game shit itself (flightsuit patch 3 v.2)

### PR DESCRIPTION
**Pull request by silicons**
UGH
the previous pr caused infinite loops

### CODE REVIEW

**Comment by optimumtact** [comment]
new word for goofcode

kevinzzzcode

**Comment by kevinz000** [comment]
:(

File: code/controllers/subsystem/processing/flightpacks.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@ var/datum/subsystem/processing/flightpacks/SSflightpacks
 	priority = 5
 	wait = 2
 	stat_tag = "FM"
-	flags = SS_NO_INIT|SS_FIRE_IN_LOBBY|SS_KEEP_TIMING
+	flags = SS_NO_INIT|SS_TICKER|SS_KEEP_TIMING
```
Comment (kevinz000) [inline code/controllers/subsystem/processing/flightpacks.dm]: This was my initial intent but now that I look at it.. this miiiight be a baaad ideaaa..... hmmmmmmm...

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -98,6 +98,8 @@
 		take_bodypart_damage(10)
 	if(iscarbon(hit_atom))
 		var/mob/living/carbon/victim = hit_atom
+		if(victim.movement_type & FLYING)
+			return
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/carbon.dm]: Prevents mobs being crashed into crashing into the wearer a second time

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 3]
```diff
@@ -711,6 +710,7 @@
 		momentum_speed_y = 1
 	else
 		momentum_speed_y = 0
+	momentum_speed = max(momentum_speed_x, momentum_speed_y)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: cheaper to do I guess?

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 4]
```diff
@@ -648,7 +655,7 @@
 	if(forced)
 		losecontrol(stun = TRUE)
 		return TRUE
-	if(abs(momentum_x) <= 20 && abs(momentum_y) <= 20)
+	if(momentum_speed < 3)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Old proc was bugged and didn't update along with regular drifting.

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 5]
```diff
@@ -626,7 +633,7 @@
 	momentum_x = 0
 	momentum_y = 0
 	if(flight)
-		disable_flight()
+		disable_flight(FALSE)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Fixes infinite loop

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 6]
```diff
@@ -611,6 +614,10 @@
 		O.take_damage(damage)
 
 /obj/item/device/flightpack/proc/losecontrol(stun = FALSE, move = TRUE)
+	if(!move)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Fixes unnecessary var checking

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 7]
```diff
@@ -484,11 +484,14 @@
 			crash_grille(S)
 		crashing = FALSE
 		return FALSE
-	else if(istype(unmovablevictim, /obj/machinery/door))
+	else if((istype(unmovablevictim, /obj/machinery/door)) && (!istype(unmovablevictim, /obj/machinery/door/poddoor)))
 		var/obj/machinery/door/D = unmovablevictim
 		if(!airlock_hit(D))
 			crashing = FALSE
 			return FALSE
+		else if(momentum_speed < 3)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Fixes crashing the moment you touch a door no matter what speed you are at

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 8]
```diff
@@ -484,11 +484,14 @@
 			crash_grille(S)
 		crashing = FALSE
 		return FALSE
-	else if(istype(unmovablevictim, /obj/machinery/door))
+	else if((istype(unmovablevictim, /obj/machinery/door)) && (!istype(unmovablevictim, /obj/machinery/door/poddoor)))
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Fixes ramming open blast doors

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 9]
```diff
@@ -409,7 +409,7 @@
 		powersetting = 1
 	momentum_gain = powersetting * 10
 	usermessage("Engine output set to [momentum_gain].")
-	momentum_drift_coeff = ((100/momentum_max)*momentum_gain*((drift_tolerance*(((drift_tolerance*2)+1)/4))/drift_tolerance)*10)
+	momentum_drift_coeff = ((momentum_gain)*(drift_tolerance*1.1))/momentum_max
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Fixes my shitty math

File: code/controllers/subsystem/processing/flightpacks.dm (hunk 1 above)
Comment (kevinz000) [inline code/controllers/subsystem/processing/flightpacks.dm]: It doesn't need to fire in lobby specifically but this'll make it way smoother to use.

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 10]
```diff
@@ -517,7 +520,7 @@
 	if(damage)
 		crash_damage(density, anchored, momentum_speed, unmovablevictim.name)
 		userknockback(density, anchored, momentum_speed, dir)
-		losecontrol(move = FALSE)
+		losecontrol(stun = FALSE, move = FALSE)
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Accidently left out an argument

File: code/controllers/subsystem/processing/flightpacks.dm [hunk 11]
```diff
@@ -2,10 +2,10 @@
 var/datum/subsystem/processing/flightpacks/SSflightpacks
 /datum/subsystem/processing/flightpacks
 	name = "Flightpack Movement"
-	priority = 5
+	priority = 30
 	wait = 2
 	stat_tag = "FM"
-	flags = SS_NO_INIT|SS_FIRE_IN_LOBBY|SS_KEEP_TIMING
+	flags = SS_NO_INIT|SS_TICKER|SS_KEEP_TIMING
```
Comment (kevinz000) [inline code/controllers/subsystem/processing/flightpacks.dm]: Makes it smoother

File: code/controllers/subsystem/processing/flightpacks.dm [hunk 12]
```diff
@@ -2,10 +2,10 @@
 var/datum/subsystem/processing/flightpacks/SSflightpacks
 /datum/subsystem/processing/flightpacks
 	name = "Flightpack Movement"
-	priority = 5
+	priority = 30
```
Comment (kevinz000) [inline code/controllers/subsystem/processing/flightpacks.dm]: Lower priority than other ticker subsystems

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 13]
```diff
@@ -54,19 +54,19 @@
...
 	var/gravity = TRUE
-	var/gravity_decay_amount = 5
+	var/gravity_decay_amount = 3
 	var/pressure = TRUE
-	var/pressure_decay_amount = 5
+	var/pressure_decay_amount = 3
 	var/pressure_threshold = 30
 	var/brake = FALSE
-	var/airbrake_decay_amount = 60
+	var/airbrake_decay_amount = 30
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Variables being halved more or less to compensate for processing speedup

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 14]
```diff
@@ -795,35 +795,45 @@
 	if(istype(I, /obj/item/weapon/stock_parts))
 		var/obj/item/weapon/stock_parts/S = I
 		if(istype(S, /obj/item/weapon/stock_parts/manipulator))
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Makes swapping parts actually eject the old part/downgrade it. Reason is before all the old parts you stuck in it were just staying in its contents without being moved out or deleted, just no longer used.

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 15]
```diff
@@ -648,7 +654,7 @@
 	if(forced)
 		losecontrol(stun = TRUE)
 		return TRUE
-	if(abs(momentum_x) <= 20 && abs(momentum_y) <= 20)
+	if(momentum_speed <= 1)
 		momentum_x = 0
```
Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: This now actually works.

**Comment by ExcessiveUseOfCobblestone** [comment]
Is this the new form of rubber duck debugging?

**Comment by kevinz000** [comment]
I'm sorry but I'm a n00b coder and my brain starts to emit smoke when I realize that I actually do have to read my own code.

**Comment by GunHog** [comment]
While you are at it, replace the coder-drawn zoom sprite with a well-made one. @WJohn has one lying around someplace!

**Pull request merged**